api/secret_test.go

```diff
@@ -0,0 +1,208 @@
+package api
+
+import (
+	"testing"
+)
+
+func TestTokenPolicies(t *testing.T) {
+	var s *Secret
+
+	// Verify some of the short-circuit paths in the function
+	if policies, err := s.TokenPolicies(); policies != nil {
+		t.Errorf("policies was not nil, got %v", policies)
+	} else if err != nil {
+		t.Errorf("err was not nil, got %v", err)
+	}
+
+	s = &Secret{}
+
+	if policies, err := s.TokenPolicies(); policies != nil {
+		t.Errorf("policies was not nil, got %v", policies)
+	} else if err != nil {
+		t.Errorf("err was not nil, got %v", err)
+	}
+
+	s.Auth = &SecretAuth{}
+
+	if policies, err := s.TokenPolicies(); policies != nil {
+		t.Errorf("policies was not nil, got %v", policies)
+	} else if err != nil {
+		t.Errorf("err was not nil, got %v", err)
+	}
+
+	s.Auth.Policies = []string{}
+
+	if policies, err := s.TokenPolicies(); policies != nil {
+		t.Errorf("policies was not nil, got %v", policies)
+	} else if err != nil {
+		t.Errorf("err was not nil, got %v", err)
+	}
+
+	s.Auth.Policies = []string{"test"}
+
+	if policies, err := s.TokenPolicies(); policies == nil {
+		t.Error("policies was nil")
+	} else if err != nil {
+		t.Errorf("err was not nil, got %v", err)
+	}
+
+	s.Auth = nil
+	s.Data = make(map[string]interface{})
+
+	if policies, err := s.TokenPolicies(); policies != nil {
+		t.Errorf("policies was not nil, got %v", policies)
+	} else if err != nil {
+		t.Errorf("err was not nil, got %v", err)
+	}
+
+	// Verify that s.Data["policies"] are properly processed
```

Comment:
I like this way of structuring bits of a test! Makes things a lot easier to understand at a glance. I might steal this :)